### ISSUE TITLE
Feature request: any(dim) all(dim)

### ISSUE THREAD

**Issue by stefdoerr** [pytorch/pytorch#4313]
Right now it seems that the `.any` and `.all` methods of `ByteTensor` don't accept a dimension and only operate on the whole array. It would be useful to be able to do these operations over a specific dimension only like `tf.reduce_all` does since now it forces me to do `x.float().sum(dim=1) == x.shape[1]`

**Comment by vishwakftw** [comment]
I believe this can be closed now.
